src/routes/HomeScreen.tsx

```diff
@@ -105,28 +85,39 @@ const HomeScreen = ({ navigation, customDisciplines }: HomeScreenPropsType): JSX
         <PlusIcon />
         <AddCustomDisciplineText>{labels.home.addCustomDiscipline}</AddCustomDisciplineText>
       </AddCustomDisciplineContainer>
-      {customDisciplines.map(customDiscipline => {
+      {customDisciplines?.map(customDiscipline => {
         return (
           <CustomDisciplineMenuItem
             key={customDiscipline}
             apiKey={customDiscipline}
             selectedId={selectedId}
             setSelectedId={setSelectedId}
             navigation={navigation}
+            refresh={refreshCustomDisciplines}
           />
         )
       })}
       <ServerResponseHandler error={error} loading={loading} refresh={refresh}>
-        <StyledList
-          data={disciplines}
-          renderItem={Item}
-          keyExtractor={item => item.id.toString()}
-          scrollEnabled={true}
-          bounces={false}
-        />
+        {disciplines?.map(item => {
+          if (item.numberOfChildren === 0) {
+            return null
+          }
+          return (
+            <MenuItem
+              key={item.id}
+              selected={item.id.toString() === selectedId}
+              title={item.title}
+              icon={item.icon}
+              onPress={() => handleNavigation(item)}>
+              <Description selected={item.id.toString() === selectedId}>
+                {item.numberOfChildren} {item.numberOfChildren === 1 ? labels.home.unit : labels.home.units}
+              </Description>
+            </MenuItem>
```

Comment:
Wouldn't it be possible to refactor `MenuItem` such that you just pass `item: DisciplineType` and it sets the props itself?

Reply:
done.